pkg/genericapiserver/config.go

```diff
@@ -301,17 +303,37 @@ func (c *Config) ApplyInsecureServingOptions(insecureServing *options.ServingOpt
 	return c
 }
 
-func (c *Config) ApplyAuthenticationOptions(o *options.BuiltInAuthenticationOptions) *Config {
+func (c *Config) ApplyAuthenticationOptions(o *options.BuiltInAuthenticationOptions) (*Config, error) {
 	if o == nil || o.PasswordFile == nil {
-		return c
+		return c, nil
 	}
 
-	if o.ClientCert != nil && c.SecureServingInfo != nil {
-		c.SecureServingInfo.ClientCA = o.ClientCert.ClientCA
+	if c.SecureServingInfo != nil {
+		if c.SecureServingInfo.ClientCA == nil {
+			c.SecureServingInfo.ClientCA = x509.NewCertPool()
```

Comment:
only initialize this if we're going to populate it... a non-nil cert pool makes the server prompt for client certs, and sending an empty CA list makes clients try to send any client cert they have